### PR TITLE
Add formatter of final period for single line docstrings and summaries

### LINKED ISSUE #10: Enforce a dot at the end of single line docstrings and summaries
This is discussed in [PEP 257](https://www.python.org/dev/peps/pep-0257/)

### PR DESCRIPTION
**Pull request by DanielNoord**
Pretty large change due to changes to the existing tests and codebase. Should probably be reviewed via the commit view.

README change explains pretty well what this does. Neat little feature for `0.3.0` I think :)

This closes #10

### CODE REVIEW

File: pydocstringformatter/formatting/__init__.py [hunk 1]
```diff
@@ -6,9 +6,11 @@
 from pydocstringformatter.formatting.formatter import (
     BeginningQuotesFormatter,
     ClosingQuotesFormatter,
+    FinalPeriodFormatter,
 )
 
 FORMATTERS: List[Formatter] = [
     BeginningQuotesFormatter(),
     ClosingQuotesFormatter(),
+    FinalPeriodFormatter(),
```
Comment (Pierre-Sassoulas) [inline pydocstringformatter/formatting/__init__.py:15]:
```suggestion
        FinalPeriodFormatter(),
        ClosingQuotesFormatter(),
```
So we manage the line break if we go further than the alloted limit with the dot ?

File: tests/data/format/final_period/function_docstrings.py [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+def func()
+    """A multi-line
+    docstring
+    """
+
+    def inner_func()
+        """A multi-line
+        docstring
+        """
+
```
Comment (Pierre-Sassoulas) [inline tests/data/format/final_period/function_docstrings.py:17]:
```suggestion

    def other_inner():
        """A docstring with a sentence ending with dot. And another one
        continuing later."""
```

I think the result should be:
```python
    def other_inner():
        """A docstring with a sentence ending with dot.

        And another one continuing later."""
```

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]

File: pydocstringformatter/formatting/__init__.py (hunk 1 above)
Comment (DanielNoord) [reply pydocstringformatter/formatting/__init__.py:15]: Hmm, but I depend on the number of line breaks to figure out what to do..

I think for now this is fine, until we have a good way of doing line wrapping? After which we'll probably need to update the order?

File: tests/data/format/final_period/function_docstrings.py (hunk 2 above)
Comment (DanielNoord) [reply tests/data/format/final_period/function_docstrings.py:17]: I didn't want to do too much with line wrapping and breaking in this and only focus on adding periods when we 99% know that they should be there (I skipped the bodies of multi line strings for the same reason). But I agree this should be done at some point.

What about `SplitSummaryAndDocstringFormatter` which splits on periods? That should be relatively easy to implement.

Comment (Pierre-Sassoulas) [reply tests/data/format/final_period/function_docstrings.py:17]: https://github.com/DanielNoord/pydocstringformatter/issues/25

File: pydocstringformatter/formatting/__init__.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply pydocstringformatter/formatting/__init__.py:15]: Fair enough !

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**